src/test/java/com/google/api/codegen/JavaSurfaceCodeGeneratorTest.java

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen;
+
+import com.google.api.codegen.gapic.MainGapicProviderFactory;
+
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Java code generator baseline tests.
+ */
+@RunWith(Parameterized.class)
+public class JavaSurfaceCodeGeneratorTest extends GapicTestBase {
+
+  public JavaSurfaceCodeGeneratorTest(
+      String name, String idForFactory, String[] gapicConfigFileNames, String snippetName) {
+    super(name, idForFactory, gapicConfigFileNames, snippetName);
+    getTestDataLocator().addTestDataSource(com.google.api.codegen.java.JavaTypeTable.class, "");
```

Comment:
Specifying the test data path this way is non-obvious, and has nothing to do with `JavaTypeTable`. I know you inherited this from the existing code, but maybe we could start moving away from this pattern?